src/test/java/dev/jbang/cli/TestRun.java

```diff
@@ -643,21 +643,6 @@ void testURLPrepareDoesNotExist() throws IOException {
 		assertThrows(ExitException.class, () -> RunContext.empty().forResource(url));
 	}
 
-	@Test
-	void testFindMain(@TempDir Path dir) throws IOException {
```

Comment:
why remove this test? shouldn't it just be changed to use a source?

Reply:
It was testing a method that was not being used anymore and was removed.

Reply:
hmm - so that is actually unrelated to the issue - just old dead code?

Reply:
Not entirely unrelated. I changed some things related to the handling of main class and noticed the unused methods.